sparta/sparta/events/PhasedSingleCycleUniqueEvent.hpp

```diff
@@ -154,6 +154,19 @@ namespace sparta
             }
         }
 
+        /**
+         * \brief Have this SingleCycleUniqueEvent precede another
+         * \param consumer The Scheduleable to follow this Scheduleable
+         * \param reason The reason for the precedence
+         *
+         *  \a this will preceed, or come before, the consumer
+         *
+         */
+        template<class ScheduleableT>
+        void precedes(ScheduleableT & consumer, const std::string & reason = "") {
+            single_cycle_event_scheduleable_.precedes(consumer);
```

Comment:
What is the purpose of ```reason```?

Reply:
Good catch -- forgot to pass it through.  Surprised the compiled didn't catch it.